### ISSUE TITLE
Valid identifiers are being marked as invalid (Go)

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#690]
(Original issue 983 created by mattholt on 2014-05-02T19:21:09.959978+00:00)

Go is defined over UTF-8, and identifiers like `世界` and `さようなら` are valid. ([See this play](http://play.golang.org/p/pSELBMujlD).) However, they're being highlighted as invalid with Pygments. [Demo](http://pygments.org/demo/360478/):

```go
package main

import "fmt"

func main() {
    世界 := "Hello, world!"
    さようなら := "Goodbye, world!"
    fmt.Println(世界)
    fmt.Println(さようなら)
}
```

The problem line is the regex on [line 2163 of compiled.py](https://bitbucket.org/birkenfeld/pygments-main/src/80ce5e90f9a894e0d94803ecdde7de61445ebfe6/pygments/lexers/compiled.py?at=default#cl-2163):

```python
# identifier
(r'[a-zA-Z_]\w*', Name.Other),
```

**Comment by Anteru** [comment]
(Original issue 983 last updated on 2014-05-20T14:42:38.694326+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by tshatch on 2014-05-04T22:47:29.674844+00:00)

Do you have a link to the Go grammar so we can know which characters are valid?

**Comment by Anteru** [comment]
(Original comment by mattholt on 2014-05-04T23:06:54.683529+00:00)

Here's the spec's definition of identifiers in Go: http://golang.org/ref/spec#Identifiers

**Comment by Anteru** [comment]
(Original comment by mwsherman on 2014-05-04T23:11:47.165297+00:00)

Here’s a regex I’ve used in my code: \p{L}[\p{L}\p{N}]*

Based on: http://www.regular-expressions.info/unicode.html#category

**Comment by Anteru** [comment]
(Original comment by tshatch on 2014-05-20T14:42:38.685955+00:00)

Go allows unicode for identifiers.

Resolves #983

→ <<cset 476d6b25042f>>
